go/exorelay/main.go

```diff
@@ -57,3 +66,42 @@ func (exoRelay *ExoRelay) Send(eventName string, payload map[string]interface{})
 
 	return websocket.Message.Send(exoRelay.socket, serializedBytes)
 }
+
+func (exoRelay *ExoRelay) receiveMessage(message structs.Message) {
+	if exoRelay.messageHandlerMapping == nil {
+		return
+	}
+	if exoRelay.messageHandlerMapping[message.Name] == nil {
+		return
+	}
+	exoRelay.messageHandlerMapping[message.Name](message.Payload)
+}
+
+func (exoRelay *ExoRelay) listenForMessages() {
+	for {
+		message, err := exoRelay.readMessage()
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			fmt.Println("Error reading message from websocket:", err)
+			break
+		} else {
+			exoRelay.receiveMessage(message)
+		}
+	}
+}
+
+func (exoRelay *ExoRelay) readMessage() (structs.Message, error) {
+	var bytes []byte
+	if err := websocket.Message.Receive(exoRelay.socket, &bytes); err != nil {
+		return structs.Message{}, err
+	}
+
+	var unmarshaled structs.Message
+	err := json.Unmarshal(bytes, &unmarshaled)
+	if err != nil {
+		return structs.Message{}, err
+	}
+
+	return unmarshaled, nil
+}
```

Comment:
The problem is that we will create an exoservice code base in the furure, and it will also need an exocommock and exorelay. 

If we can make it so that the exoservice code base can contain everything, and export a CLI tool and the ExoRelay library and the ExocomMock library, that would solve our problem. 

Meaning people must be able to do this:

Import "github.com/Originate/exocom/go/exoservice/exorelay" and get the ExoRelay. 

Import "github.com/Originate/exocom/go/exoservice/exocommock" and get the exocommock library.